client/QuickStart/Program.cs

```diff
@@ -0,0 +1,767 @@
+    public class ConfigJson
+    {
+        public bool? UseExistingTable = null;
+        public string DatabaseName = null;
+        public string TableName = null;
+        public string TableSchema = null;
+        public string KustoUri = null;
+        public string IngestUri = null;
+        public string CertificatePath = null;
+        public string CertificatePassword = null;
+        public string ApplicationId = null;
+        public string TenantId = null;
+        public List<Dictionary<string, string>> Data = null;
+        public bool? AlterTable = null;
+        public bool? QueryData = null;
+        public bool? IngestData = null;
+    }
+
+
+    /// <summary>
+    /// The quick start application is a self-contained and runnable example script that demonstrates authenticating,
+    /// connecting to, administering, ingesting data into and querying Azure Data Explorer using the azure-kusto C# SDK.
+    /// You can use it as a baseline to write your own first kusto client application, altering the code as you go,
+    /// or copy code sections out of it into your app.
+    /// Tip: The app includes comments with tips on recommendations, coding best practices, links to reference materials
+    /// and recommended TO DO changes when adapting the code to your needs.
+    /// </summary>
+    public static class KustoSampleApp
+    {
+        //########################### Constants Start ###########################// 
+
+        // TODO (config - optional): Change the authentication method from "User Prompt" to any of the other options
+        //  Some of the auth modes require additional environment variables to be set in order to work
+        // (see usage in generate_connection_string below).
+        //  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+        private const string
+            AuthenticationMode = "UserPrompt"; // Options: (UserPrompt|ManagedIdentity|AppKey|AppCertificate)
+
+        // TODO (config - optional): Toggle to False to execute this script "unattended"
+        private const bool WaitForUser = true;
+
+        // TODO (config):
+        // If this quickstart app was downloaded from OneClick, kusto_sample_config.json should be pre-populated
+        // with your cluster's details.
+        // If this quickstart app was downloaded from GitHub, edit kusto_sample_config.json and modify the cluster URL
+        // and database fields appropriately.
+        private const string ConfigFileName = @"kusto_sample_config.json";
+        private const int WaitForIngestSeconds = 20;
+        private static int _step = 1;
+
+        private const string BatchingPolicy = "{ 'MaximumBatchingTimeSpan': '00:00:10', 'MaximumNumberOfItems': 500, " +
+                                              "'MaximumRawDataSizeMB': 1024 }";
+        //########################### Constants End ###########################// 
+
+
+        public static async Task Main()
```

Comment:
I would keep the same structure for the main function as in Java and Python
It allows the reader to quickly understand what the application does even if he/she does not understand how it is technically done

Reply:
@AsafMah Merging both of your comments on this - This structure is the structure that appears in python - a main function summoning Start. i do agree its redundant and not really needed. Given all of this, what would you have me do? stick to our Python format, or remove the Start and merge it into Main?

Reply:
Yes, but also yochai means to inline `BasicTableCommands`, `IngestionHandler` and `ExecuteValidationQueries` into Main, so it's easier to see the flow, like in python/java

Reply:
as in - not to encapsulate the code fragments into building blocks, but to present it in Start so the user can see it all there? (not in main btw, unless you meant something else)